synapseclient/models/mixins/json_schema.py

```diff
@@ -0,0 +1,471 @@
+    """A list of all validation messages."""
+
+    validation_exception: ValidationException
+    """The validation exception details."""
+
+
+@dataclass
+class JSONSchemaDerivedKeys:
+    """Represents the response for derived JSON schema keys."""
+
+    keys: List[str]
+    """A list of derived keys for the entity."""
+
+
+@async_to_sync
+class BaseJSONSchema(BaseJSONSchemaProtocol):
+    """
+    Mixin class to provide JSON schema functionality.
+    This class is intended to be used with classes that represent Synapse entities.
+    It provides methods to bind, delete, and validate JSON schemas associated with the entity.
+    """
+
+    id: Optional[str] = None
+
+    async def bind_schema_async(
+        self,
+        json_schema_uri: str,
+        *,
+        enable_derived_annos: bool = False,
+        synapse_client: Optional["Synapse"] = None,
+    ) -> JSONSchemaBinding:
+        """
+        Bind a JSON schema to the entity.
+
+        Args:
+            json_schema_uri (str): The URI of the JSON schema to bind to the entity.
+            enable_derived_annos (bool, optional): If true, enable derived annotations. Defaults to False.
+            synapse_client (Optional[Synapse], optional): The Synapse client instance. If not provided,
+                the last created instance from the Synapse class constructor will be used.
+
+        Returns:
+            JSONSchemaBinding: An object containing details about the JSON schema binding.
+        """
+        response = await bind_json_schema_to_entity(
+            synapse_id=self.id,
+            json_schema_uri=json_schema_uri,
+            enable_derived_annos=enable_derived_annos,
+            synapse_client=synapse_client,
+        )
+        json_schema_version = response.get("jsonSchemaVersionInfo", {})
+        return JSONSchemaBinding(
+            json_schema_version_info=JSONSchemaVersionInfo(
+                organization_id=json_schema_version.get("organizationId", ""),
+                organization_name=json_schema_version.get("organizationName", ""),
+                schema_id=json_schema_version.get("schemaId", ""),
+                id=json_schema_version.get("$id", ""),
+                schema_name=json_schema_version.get("schemaName", ""),
+                version_id=json_schema_version.get("versionId", ""),
+                semantic_version=json_schema_version.get("semanticVersion", ""),
+                json_sha256_hex=json_schema_version.get("jsonSHA256Hex", ""),
+                created_on=json_schema_version.get("createdOn", ""),
+                created_by=json_schema_version.get("createdBy", ""),
+            ),
+            object_id=response.get("objectId", ""),
+            object_type=response.get("objectType", ""),
+            created_on=response.get("createdOn", ""),
+            created_by=response.get("createdBy", ""),
+            enable_derived_annotations=response.get("enableDerivedAnnotations", False),
+        )
```

Comment:
I'd expect that the default value is `None` rather than an empty string. An empty string has a different meaning to `None`, and if Synapse isn't returning a value `None` is the correct meaning